packages/grid/x-data-grid/src/tests/quickFiltering.DataGrid.test.tsx

```diff
@@ -607,4 +607,20 @@ describe('<DataGrid /> - Quick Filter', () => {
     await sleep(debounceMs * 2);
     expect(searchBox.value).to.equal('abc');
   });
+
+  // https://github.com/mui/mui-x/issues/9666
+  it('should not fail when the data changes', () => {
```

Comment:
For this test to fail without a fix, it should override the default `getApplyQuickFilterFn`. Otherwise, it works fine.
You can check this demo and try removing the `getApplyQuickFilterFn` override https://codesandbox.io/s/practical-pasteur-hlknzw

Reply:
Right, I've updated, it fails on master now.